include/pagmo/problems/golomb_ruler.hpp

```diff
@@ -0,0 +1,216 @@
+ * two of its marks is its length. Translation and reflection of a Golomb ruler are considered trivial, so the smallest
+ * mark is customarily put at 0 and the next mark at the smaller of its two possible values.  There is no requirement
+ * that a Golomb ruler be able to measure all distances up to its length, but if it does, it is called a perfect Golomb
+ * ruler.
+ *
+ * A Golomb ruler is optimal if no shorter Golomb ruler of the same order exists. Creating Golomb rulers is easy,
+ * but finding the optimal Golomb ruler (or rulers) for a specified order is computationally very challenging.
+ *
+ * This UDP represents the problem of finding an optimal Golomb ruler of a given order \f$ n\f$. A maximal distance \f$
+ * l_{max}\f$ between consecutive marks is also specified to make the problem representation possible. The resulting
+ * optimization problem is an integer programming problem with one equality constraint.
+ *
+ * In this UDP, the decision vector is \f$ x=[d_1, d_2, d_{n-1}]\f$, where the distances between consecutive ticks are
+ * indicated with \f$ d_i\f$. The ticks on the ruler can then be reconstructed as \f$ a_0 = 0\f$, \f$ a_i = sum_{j=1}^i
+ * d_i, i=1 .. n-1\f$
+ *
+ * Its formulation can thus be written as:
+ *
+ * \f[
+ *   \begin{array}{rl}
+ *   \mbox{find:} & 1 \le d_i \le l_{max}, \forall i=1..n-1 \\
+ *   \mbox{to minimize: } & \sum_i d_i  \\
+ *   \mbox{subject to:} & |a_i-a_j| \neq |a_l - a_m|, \forall (\mbox{distinct}) i,j,l,m \in [0, n]
+ * \end{array}
+ * \f]
+ *
+ * We transcribe the constraints as one single equality constraint: \f$ c = 0 \f$ where \f$ c\f$ is the count of
+ * repeated distances.
+ *
+ * See: https://en.wikipedia.org/wiki/Golomb_ruler
+ *
+ */
+class golomb_ruler
+{
+public:
+    /// Constructor from ruler order and maximal consecutive ticks distance
+    /**
+     * Constructs a Golomb Ruler problem.
+     *
+     * @param order ruler order.
+     * @param upper_bound maximum distance between consecutive ticks.
+     *
+     * @throw std::invalid_argument if \p order is < 2 or \p upper_bound is < 1.
+     * @throw std::overflow_error if \p upper_bound is too large.
+     */
+    golomb_ruler(unsigned order = 3u, unsigned upper_bound = 10) : m_order(order), m_upper_bound(upper_bound)
+    {
+        if (order < 2u) {
+            pagmo_throw(std::invalid_argument, "Golomb ruler problem must have at least order 2, while "
+                                                   + std::to_string(order) + " was requested.");
+        }
+        if (upper_bound < 2u) {
+            pagmo_throw(
+                std::invalid_argument,
+                "The upper bound for the maximum distance between consecutive ticks has to be at least 2, while "
+                    + std::to_string(upper_bound) + " was requested.");
+        }
+        // Overflow can occur when evaluating the fitness later if the upper_bound is too large.
+        if (static_cast<double>(upper_bound) * (order - 1u)
```

Comment:
This overflow check should not involve floating point arithmetics. You can rewrite it as

```c++

if (upper_bound > std::numeric_limits<unsigned>::max() /  (order - 1u)) {
    pagmo_throw(...);
}

```